Src/Shaders/PBREffect.fx

```diff
@@ -0,0 +1,284 @@
+    vout.current.TangentWS = normalize(mul(tangent, WorldInverseTranspose));
+
+    vout.prevPosition = mul(vin.Position, PrevWorldViewProj);
+
+    return vout;
+}
+
+
+// Pixel shader: pbr (constants) + image-based lighting
+float4 PSConstant(PSInputPixelLightingTxTangent pin) : SV_Target0
+{
+    // vectors
+    const float3 V = normalize(EyePosition - pin.PositionWS.xyz);   // view vector
+    const float3 N = normalize(pin.NormalWS);                       // surface normal
+    const float AO = 1;                                             // ambient term
+
+    float3 color = LightSurface(V, N, 3,
+        LightColor, LightDirection,
+        ConstantAlbedo, ConstantRoughness, ConstantMetallic, AO);
+
+    return float4(color, Alpha);
+}
+
+
+// Pixel shader: pbr (textures) + image-based lighting
+float4 PSTextured(PSInputPixelLightingTxTangent pin) : SV_Target0
+{
+    const float3 V = normalize(EyePosition - pin.PositionWS.xyz); // view vector
+    const float3 L = normalize(-LightDirection[0]);               // light vector ("to light" opposite of light's direction)
+
+    // Before lighting, peturb the surface's normal by the one given in normal map.
+    float3 localNormal = BiasX2(NormalTexture.Sample(SurfaceSampler, pin.TexCoord).xyz);
+    float3 N = PeturbNormal(localNormal, pin.NormalWS, pin.TangentWS);
+
+    // Get albedo
+    float4 albedo = AlbedoTexture.Sample(SurfaceSampler, pin.TexCoord);
+
+    // Get roughness, metalness, and ambient occlusion
+    float3 RMA = RMATexture.Sample(SurfaceSampler, pin.TexCoord);
+
+    // glTF2 defines metalness as B channel, roughness as G channel, and occlusion as R channel
+
+    // Shade surface
+    float3 color = LightSurface(V, N, 3, LightColor, LightDirection, albedo.rgb, RMA.g, RMA.b, RMA.r);
+
+    return float4(color, albedo.w * Alpha);
+}
+
+
+// Pixel shader: pbr (textures) + emissive + image-based lighting
+float4 PSTexturedEmissive(PSInputPixelLightingTxTangent pin) : SV_Target0
+{
+    const float3 V = normalize(EyePosition - pin.PositionWS.xyz); // view vector
+    const float3 L = normalize(-LightDirection[0]);               // light vector ("to light" opposite of light's direction)
+
+    // Before lighting, peturb the surface's normal by the one given in normal map.
+    float3 localNormal = BiasX2(NormalTexture.Sample(SurfaceSampler, pin.TexCoord).xyz);
+    float3 N = PeturbNormal(localNormal, pin.NormalWS, pin.TangentWS);
+
+    // Get albedo
+    float4 albedo = AlbedoTexture.Sample(SurfaceSampler, pin.TexCoord);
+
+    // Get roughness, metalness, and ambient occlusion
+    float3 RMA = RMATexture.Sample(SurfaceSampler, pin.TexCoord);
+
+    // glTF2 defines metalness as B channel, roughness as G channel, and occlusion as R channel
+
+    // Shade surface
+    float3 color = LightSurface(V, N, 3, LightColor, LightDirection, albedo.rgb, RMA.g, RMA.b, RMA.r);
+
+    color += EmissiveTexture.Sample(SurfaceSampler, pin.TexCoord).rgb;
```

Comment:
glTF2 supports an emissive channel which was not in the original SimplePBR12 sample. I've added here as an option.